### DISCUSSION TITLE
What do @request.query and @request.data return?

### DISCUSSION THREAD

**Discussion by dotargz** [pocketbase/pocketbase#323] [Q&A] [answered]
I can't seem to find the relevant documentation on what these filter parameters return.

Additionally, how could I allow an update action if it only updates a certain field, otherwise disallow it?

**Comment by ganigeorgiev** [comment, marked as answer]
`@request.query.*` - returns the value of any query parameter sent with the current request. For example, if your request is:
```
GET /api/collections/example/records?test=123
```
then `@request.query.test` will return `123`, while `@request.query.missing` will resolve to `NULL`
(_when doing `=` or `!=` comparision `NULL` values are normalized to empty string, so both `null = null` and `null = ""` are true_).

`@request.data.*` is similar to the above, but instead of query parameter values it works with the request body.

---

> Additionally, how could I allow an update action if it only updates a certain field, otherwise disallow it?

It depends to what you want to compare the sent value, but usually you could use the `@request.data.*`. For example, lets say that your field is named `title` and you want to check whether the user has sent any title with the request:
```
@request.data.title != ""
```

If  you need more advanced customizations and filters, you could always use PocketBase as framework and extend it default behavior via the available event hooks - https://pocketbase.io/docs/event-hooks/#record-hooks.

**Reply by dotargz** [reply]
I am currently unable to use Pocketbase as a framework due to my limited experience in Go, but I am wondering if there is a way to allow an update action if it **only** updates a certain field, otherwise disallow it. Meaning that the user is only allowed to edit that one field, and none of the others. I understand if this is not possible, but it would be nice if it was :D

**Reply by ganigeorgiev** [reply]
> Meaning that the user is only allowed to edit that one field, and none of the others.

This is not possible, at least not directly. I can think of only 2 alternatives:
- move the fields that shouldn't be edited by the user in a new collection and reference it with a relation field
- OR you can type each field in the update api rule to ensure that it doesn't change. For example lets say that you have 3 fields - `title`, `description`, `status` and you want the user to be able to change only the `status`  field, then the following rule should be sufficient (you still need to send the original record data as body):
```
title = @request.data.title && description = @request.data.description
```
_(which translates to - the new title and description must match with the already set values)_
